interfaces/IWearableTargets/hde/interfaces/IWearableTargets.h

```diff
@@ -0,0 +1,136 @@
+
+        mutable std::mutex mutex;
+
+        WearableSensorTarget(wearable::WearableName wearableName_,
+                            ModelLinkName modelLinkName_,
+                            hde::KinematicTargetType targetType_)
+                            : wearableName(wearableName_)
+                            , modelLinkName(modelLinkName_)
+                            , targetType(targetType_)
+        {
+            position.zero();
+            rotation.Identity();
+            linearVelocity.zero();
+            angularVelocity.zero();
+            positionScaleFactor(0) = 1;
+            positionScaleFactor(1) = 1;
+            positionScaleFactor(2) = 1;
+            clearCalibrationMatrices();
+        };
+
+        void clearCalibrationMatrices()
+        {
+            std::lock_guard<std::mutex> lock(mutex);
+            calibrationWorldToMeasurementWorld = iDynTree::Transform::Identity();
+            calibrationMeasurementToLink = iDynTree::Transform::Identity();
+        };
+
+        iDynTree::Vector3 getCalibratedPosition()
+        {
+            std::lock_guard<std::mutex> lock(mutex);
+            positionInWorld = iDynTree::Position(calibrationMeasurementToLink.getPosition()).changeCoordinateFrame(rotation) + iDynTree::Position(position).changeCoordinateFrame(calibrationWorldToMeasurementWorld.getRotation()) + calibrationWorldToMeasurementWorld.getPosition();
+            // scale position
+            positionScaled.setVal(0, positionInWorld.getVal(0) * positionScaleFactor(0));
+            positionScaled.setVal(1, positionInWorld.getVal(1) * positionScaleFactor(1));
+            positionScaled.setVal(2, positionInWorld.getVal(2) * positionScaleFactor(2));
+
+            return positionScaled;
+        };
+
+        iDynTree::Rotation getCalibratedRotation()
+        {
+            std::lock_guard<std::mutex> lock(mutex);
+            return calibrationWorldToMeasurementWorld.getRotation() * rotation * calibrationMeasurementToLink.getRotation();
+        };
+
+        iDynTree::Vector3 getCalibratedLinearVelocity()
+        {
+            std::lock_guard<std::mutex> lock(mutex);
+            linearVelocityInWorld = iDynTree::LinearMotionVector3(linearVelocity).changeCoordFrame(calibrationWorldToMeasurementWorld.getRotation());
+            // scale linear velocity
+            linearVelocityScaled.setVal(0, linearVelocityInWorld.getVal(0) * positionScaleFactor(0));
+            linearVelocityScaled.setVal(1, linearVelocityInWorld.getVal(1) * positionScaleFactor(1));
+            linearVelocityScaled.setVal(2, linearVelocityInWorld.getVal(2) * positionScaleFactor(2));
+            return linearVelocityScaled;
+        };
+
+        iDynTree::Vector3 getCalibratedAngularVelocity()
+        {
+            std::lock_guard<std::mutex> lock(mutex);
+            return iDynTree::AngularMotionVector3(angularVelocity).changeCoordFrame(calibrationWorldToMeasurementWorld.getRotation());
```

Comment:
Here we should take into consideration also the offset applied in the right calibration transform